src/js/components/modals/ServiceFormModal.js

```diff
@@ -135,6 +136,13 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    let {state, props} = this;
+    return props.open !== nextProps.open ||
+      state.jsonMode !== nextState.jsonMode ||
+      state.errorMessages !== nextState.errorMessage;
+  }
```

Comment:
could there be a case where the user didn't fix the error message and we need to re-render it?